lib/src/dartdoc_options.dart

```diff
@@ -1617,6 +1620,13 @@ Future<List<DartdocOption>> createDartdocOptions() async {
             'exist. Executables for different platforms are specified by '
             'giving the platform name as a key, and a list of strings as the '
             'command.'),
+    DartdocOptionArgOnly<String>("templatesDir", null, isDir: true, mustExist: true, hide: true,
+        help: 'Path to a directory containing templates to use instead of the default ones. '
+            'Directory must contain an html file for each of the following: 404error, category, '
+            'class, constant, constructor, enum, function, index, library, method, mixin, '
+            'property, top_level_constant, top_level_property, typedef. Partial templates are '
+            'supported; they must begin with an underscore, and references to them must omit the '
+            'leading underscore (e.g. use {{>foo}} to reference the partial template _foo.html).'),
```

Comment:
It would be good to have a check to see if all the expected templates are in the directory, and fail with the a message that lets the user know what's missing. The template files can change (working on one such change now), and it will be helpful to flag this with a check rather than a failure/stack trace.

Reply:
Added a check (and test), but let me know if you want me to go about it in a different way.